From 3601555139734283f9618d4a041300e0cbe0654b Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 23 Jan 2025 14:37:37 +0100
Subject: [PATCH] drm/panel: himax-hx8394: remove read ID sequence

This sequence is not necessary for the current use of the driver &
it generates crashes if the panel is not detected.

Change-Id: Iac1980bd6586827a8a5a7fa2b4df4eabc9883359
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/431066
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/panel/panel-himax-hx8394.c | 20 --------------------
 1 file changed, 20 deletions(-)

--- a/drivers/gpu/drm/panel/panel-himax-hx8394.c
+++ b/drivers/gpu/drm/panel/panel-himax-hx8394.c
@@ -221,8 +221,6 @@ static const struct hx8394_panel_desc hs
 static int rocktech_rk055mhd042a0_init_sequence(struct hx8394 *ctx)
 {
 	struct mipi_dsi_device *dsi = to_mipi_dsi_device(ctx->dev);
-	u8 id1, id2, id3;
-	int ret;
 
 	mipi_dsi_dcs_write_seq(dsi, HX8394_CMD_SETEXTC,
 			       0xFF, 0x83, 0x94);
@@ -294,24 +292,6 @@ static int rocktech_rk055mhd042a0_init_s
 
 	mipi_dsi_dcs_write_seq(dsi, MIPI_DCS_SET_ADDRESS_MODE, FH);
 
-	ret = mipi_dsi_dcs_read(dsi, HX8394_CMD_READ_ID1, &id1, 1);
-	if (ret < 0) {
-		dev_err(ctx->dev, "could not read MTP ID1\n");
-		return ret;
-	}
-	ret = mipi_dsi_dcs_read(dsi, HX8394_CMD_READ_ID2, &id2, 1);
-	if (ret < 0) {
-		dev_err(ctx->dev, "could not read MTP ID2\n");
-		return ret;
-	}
-	ret = mipi_dsi_dcs_read(dsi, HX8394_CMD_READ_ID3, &id3, 1);
-	if (ret < 0) {
-		dev_err(ctx->dev, "could not read MTP ID3\n");
-		return ret;
-	}
-
-	dev_info(ctx->dev, "MTP ID manufacturer: %02x version: %02x driver: %02x\n", id1, id2, id3);
-
 	return 0;
 }
 
